edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -834,10 +834,11 @@ impl ModuleRuntime for DockerModuleRuntime {
                     c.name(),
                     wait_before_kill,
                 )
-                .or_else(|err| match (&err).kind() {
-                    ErrorKind::NotFound(_) | ErrorKind::NotModified => Ok(()),
-                    _ => Err(err),
-                })
+                .or_else(|err| match Fail::find_root_cause(&err).downcast_ref::<ErrorKind>() {
+                        Some(ErrorKind::NotFound(_)) | Some(ErrorKind::NotModified) => Ok(()),
```

Comment:
Would there be any other error code? Sometimes docker container can hang that the runtime is not able to stop it; or docker itself hangs.

Reply:
After looking through the errors returned from stop, I think these are the two we want to ignore. Other errors signal something more sinister, and I think we want to fail in those cases.